stock_picking_group_by_partner_by_carrier/models/stock_rule.py

```diff
@@ -2,12 +2,26 @@
 # Copyright 2020 Jacques-Etienne Baudoux (BCIM) <je@bcim.be>
 # License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl.html).
 
-from odoo import models
+from odoo import fields, models
 
 
 class StockRule(models.Model):
     _inherit = "stock.rule"
 
+    propagate_original_group = fields.Boolean(
+        help="Propagate the original group on the created moves. This allows to "
+        "prevent merging pulled moves. This allows to cancel a SO without "
+        "canceling pulled moves from other SO (as we ensure they are not "
+        "merged). If your warehouse is configured to make the delivery in two "
+        "steps, you should not propagate the original group on the picking if your"
+        "pick picking type is also configured to group pickings. In this way, "
+        "the generated moves for the same product but for different sales orders "
+        "will be merged into one unique move. If you propagate the original group "
+        "on the picking, the moves will not be merged and you will have one move "
+        "per sale order.",
```

Comment:
I would not talk about configuring merging on non outgoing moves as this should not be possible. The option is only visible on outgoing picking types. We should add a constrain to make it more explicit.
```suggestion
        "merged). You likely need this option if you don't propagate the merged procurement group to the pulled moves.",
```